### ISSUE TITLE
Unable to specify MediaType when retrieving pages of a Staff's Media

### ISSUE THREAD

**Issue by danejur** [AniList/ApiV2-GraphQL-Docs#22]
Right now, if I want to view the Media that a particular Staff person was involved in, I can use this query:
```query ($id: Int, $page: Int, $perPage: Int) {
  Data: Staff(id: $id) {
    staffMedia(page: $page, perPage: $perPage) {
      pageInfo {
        hasNextPage
      }
      edges {
        node {
          id
        }
      }
    }
  }
}
```
This returns results for both `ANIME` and `MANGA`. This makes lazy-loading these results a challenge, as MediaType checking must be done at load-time, and there is a possibility that given page won't contain any records with the MediaType desired.

A preferable solution would be to allow `staffMedia` to take the parameter `type`, similar to the parameter on the standard `Media` query. An alternative would be to allow the `Media` query to take the parameter `staffId`, indicating that all results for the given query contain the supplied staffId as a staff member of that media.

**Comment by danejur** [comment]
I'll go ahead and add that this same issue exists when trying to query a Character's Media.

**Comment by joshstar** [comment]
Good point, I've added type arguments to the Staff and Character media fields.
